app/assets/javascripts/static/GalleryView.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * The Interesting view.
+ */
+define([
+  'jquery',
+  'backbone',
+  'handlebars',
+  'mps',
+  'chosen',
+  'simplePagination',
+  'static/helpers/galleryHelper',
+  'text!static/templates/gallery.handlebars',
+], function($,Backbone,Handlebars,mps,chosen,simplePagination,galleryHelper,tpl) {
+
+  'use strict';
+
+  var GalleryCollection = Backbone.Collection.extend({
+
+    comparator: function(item) {
+      return parseInt(item.get("order"))
+    },
+
+    getPaginatedCollection: function(currentPage,itemsOnPage,filter) {
+      var filteredCollection = (filter == 'all') ? this.toJSON() : this.filter(filter);
+      return filteredCollection.slice(currentPage*itemsOnPage, (currentPage*itemsOnPage) + itemsOnPage)
+    },
+
+    getCount: function(filter) {
+      var filteredCollection = (filter == 'all') ? this.toJSON() : this.filter(filter);
+      return filteredCollection.length;
+    },
+
+    filter: function(filter) {
+      return _.compact(_.map(this.toJSON(), function(v){
+        return (v.filter == filter) ? v : null
+      }))
```

Comment:
This code is fine, though you can be a bit more idiomatic with regards to using Backbone's built in functions by replacing your custom filter function with something like: `this.where({filter: filter})`. This returns models, so if you want JS objects, you could use something like `_.invoke(this.where({filter: filter}), 'toJSON')`.

The more work we make Backbone do, and the less code we write, the better! :joy: